pilot/pkg/model/envoyfilter.go

```diff
@@ -52,7 +52,22 @@ func convertToEnvoyFilterWrapper(local *Config) *EnvoyFilterWrapper {
 		}
 		// there wont be an error here because validation catches mismatched types
 		cpw.Value, _ = config.BuildXDSObjectFromStruct(cp.ApplyTo, cp.Patch.Value)
-		out.ConfigPatches = append(out.ConfigPatches, cpw)
+		if cp.Match == nil {
+			// create a match all object
+			cpw.Match = &networking.EnvoyFilter_EnvoyConfigObjectMatch{Context: networking.EnvoyFilter_ANY}
+		}
+		if _, exists := out.Patches[cp.ApplyTo]; !exists {
+			out.Patches[cp.ApplyTo] = make([]*EnvoyFilterConfigPatchWrapper, 0)
+		}
+		if cpw.Operation == networking.EnvoyFilter_Patch_INSERT_AFTER ||
+			cpw.Operation == networking.EnvoyFilter_Patch_INSERT_BEFORE {
+			// insert_before or after is applicable only for network filter and http filter
+			// convert the rest to add
+			if cpw.ApplyTo != networking.EnvoyFilter_HTTP_FILTER && cpw.ApplyTo != networking.EnvoyFilter_NETWORK_FILTER {
```

Comment:
should we just reject these in validation?

Reply:
hmm.. thats also an option